docs/contributors/code/coding-guidelines.md

```diff
@@ -486,13 +403,51 @@ const privateSettings = [
 ];
 ```
 
-#### Experimental block.json and theme.json APIs
+#### Private block.json and theme.json APIs
 
 As of today, there is no way to restrict the `block.json` and `theme.json` APIs
-to the Gutenberg codebase. In the future, however, the new `__experimental` APIs
+to the Gutenberg codebase. In the future, however, the new private APIs
 will only apply to the core WordPress blocks and plugins and themes will not be
 able to access them.
 
+#### Inline small actions in thunks
+
+Finally, instead of introducing a new action creator, consider using a [thunk](/docs/how-to-guides/thunks.md):
+
+```js
+export function toggleFeature( scope, featureName ) {
+	return function ( { dispatch } ) {
+		dispatch( { type: '__private_BEFORE_TOGGLE' } );
+		// ...
+	};
+}
+```
+
+### Exposing private APIs publicly
```

Comment:
Do we really need this part of the document, seems a bit redundant with plugin only APIs. I don't feel strongly though.